### ISSUE TITLE
Model management and fine-tuning

### ISSUE THREAD

**Issue by leoll2** [open-edge-platform/training_extensions#4528]
One of the prominent features of Geti Tune is the ability to fine-tune models on the data collected by a source.

## Requirements

1. The user must be able to train models on CPU, GPU and XPU
2. The user must be able to choose the architecture to train, among the supported ones
3. The training dataset should consist of media with manual annotations or accepted predictions
4. The user must be able to monitor the progress of the training
5. The user must be able to configure the main hyperparameters of the model
6. The user must be able to configure the data augmentation settings
7. The user must be able to customize the subset split
8. The user must be able to manually activate a trained model
9. The user must be able to browse the dataset and labels used to train a specific model
